lib/IDE/Refactoring.cpp

```diff
@@ -1800,6 +1800,48 @@ collectAvailableRefactoringsAtCursor(SourceFile *SF, unsigned Line,
   return collectAvailableRefactorings(SF, Tok, Scratch, /*Exclude rename*/false);
 }
 
+static bool performCasesExpansionInSwitchStmt(SwitchStmt *SwitchS,
+                                              DiagnosticEngine &DiagEngine,
+                                              SourceLoc ExpandedStmtLoc,
+                                              EditorConsumerInsertStream &OS
+                                              ) {
+  // To find the subject enum decl for this switch statement; if failing,
+  // issue errors.
+  EnumDecl *SubjectED = nullptr;
+  if (auto SubjectTy = SwitchS->getSubjectExpr()->getType()) {
+    SubjectED = SubjectTy->getAnyNominal()->getAsEnumOrEnumExtensionContext();
+  }
+  if (!SubjectED) {
+    DiagEngine.diagnose(ExpandedStmtLoc, diag::no_subject_enum);
```

Comment:
This... isn’t correct.  For one you can switch over arbitrary expressions, including tuples, so `isApplicable` should probably be updated to take into account what expressions this action can handle.

Reply:
It's actually how Expand Default is implemented right now, before this PR is introduced. I've just moved the code and generalized it to add handling when `switch` is selected. You're right — I will fix `isApplicable` in both `Expand Default` and `Expand Switch Cases`. Thanks!

Reply:
Oh, sorry that wasn't your bug.  Syntactic expansion here is a very nuanced topic for me 😄

Reply:
It's great that you've spotted this. There are couple other places like that in `Expand Default` `performChange` — where we could move the checking to `isApplicable`. It's a perfect moment to... refactor it :)